plugins/cryptokiplugin/plugin.cpp

```diff
@@ -0,0 +1,227 @@
+/*
+ * Copyright (C) 2017 Jolla Ltd.
+ * Contact: Chris Adams <chris.adams@jollamobile.com>
+ * All rights reserved.
+ * BSD 3-Clause License, see LICENSE *
+ *
+ * Copyright (C) 2017 Open Mobile Platform LLC.
+ * Contact: Denis Semakin <d.semakin@omprussia.ru>
+ * All rights reserved.
+ */
+
+#include "plugin.h"
+
+#include "Crypto/key.h"
+#include "Crypto/certificate.h"
+
+#include <QtCore/QByteArray>
+#include <QtCore/QMap>
+#include <QtCore/QVector>
+#include <QtCore/QString>
+#include <QtCore/QCryptographicHash>
+
+Q_PLUGIN_METADATA(IID Sailfish_Crypto_CryptoPlugin_IID)
+
+Sailfish::Crypto::Daemon::Plugins::CryptokiPlugin::CryptokiPlugin(QObject *parent)
+    : Sailfish::Crypto::CryptoPlugin(parent)
+{
+    //osslevp_init();
+    //TODO: WIP
+}
+
+Sailfish::Crypto::Daemon::Plugins::CryptokiPlugin::~CryptokiPlugin()
+{
+}
+
+QVector<Sailfish::Crypto::Key::Algorithm>
+Sailfish::Crypto::Daemon::Plugins::CryptokiPlugin::supportedAlgorithms() const
+{
+    QVector<Sailfish::Crypto::Key::Algorithm> retn;
+    retn.append(Sailfish::Crypto::Key::Aes256);
+    return retn;
+}
+
+QMap<Sailfish::Crypto::Key::Algorithm, Sailfish::Crypto::Key::BlockModes>
+Sailfish::Crypto::Daemon::Plugins::CryptokiPlugin::supportedBlockModes() const
+{
+    QMap<Sailfish::Crypto::Key::Algorithm, Sailfish::Crypto::Key::BlockModes> retn;
+    retn.insert(Sailfish::Crypto::Key::Aes256, Sailfish::Crypto::Key::BlockModeCBC);
+    return retn;
+}
+
+QMap<Sailfish::Crypto::Key::Algorithm, Sailfish::Crypto::Key::EncryptionPaddings>
+Sailfish::Crypto::Daemon::Plugins::CryptokiPlugin::supportedEncryptionPaddings() const
+{
+    QMap<Sailfish::Crypto::Key::Algorithm, Sailfish::Crypto::Key::EncryptionPaddings> retn;
+    retn.insert(Sailfish::Crypto::Key::Aes256, Sailfish::Crypto::Key::EncryptionPaddingNone);
+    return retn;
+}
+
+QMap<Sailfish::Crypto::Key::Algorithm, Sailfish::Crypto::Key::SignaturePaddings>
+Sailfish::Crypto::Daemon::Plugins::CryptokiPlugin::supportedSignaturePaddings() const
+{
+    QMap<Sailfish::Crypto::Key::Algorithm, Sailfish::Crypto::Key::SignaturePaddings> retn;
+    retn.insert(Sailfish::Crypto::Key::Aes256, Sailfish::Crypto::Key::SignaturePaddingNone);
+    return retn;
+}
+
+QMap<Sailfish::Crypto::Key::Algorithm, Sailfish::Crypto::Key::Digests>
+Sailfish::Crypto::Daemon::Plugins::CryptokiPlugin::supportedDigests() const
+{
+    QMap<Sailfish::Crypto::Key::Algorithm, Sailfish::Crypto::Key::Digests> retn;
+    retn.insert(Sailfish::Crypto::Key::Aes256, Sailfish::Crypto::Key::DigestSha256);
```

Comment:
API question: is the digest required for encrypt/decrypt operations or not?  I wasn't sure, so included a digest parameter for those operations, but perhaps that doesn't make sense.  Maybe digest is only needed for sign/verify operations?